src/DataProtection/DataProtection/test/Microsoft.AspNetCore.DataProtection.Tests/KeyManagement/DefaultKeyResolverTests.cs

```diff
@@ -291,9 +292,180 @@ public void ResolveDefaultKeyPolicy_OlderUnpropagatedKeyPreferred()
         Assert.False(resolution.ShouldGenerateNewKey);
     }
 
+    [Fact]
```

Comment:
Should we add a test here that validates the `MaximumDefaultKeyResolverRetries` is respected? It doesn't seem like that is covered directly anywhere.

Reply:
I think it's covered [here](https://github.com/dotnet/aspnetcore/pull/54711/files/c71056cce15e145ba8a5a419f61b41dd0de17a7c#diff-45ce103f478a21db749fbe044d7bfac2f38aaebcf9e7c95af02937bab7781443R406).